apis/v1beta1/httproute_types.go

```diff
@@ -495,11 +495,13 @@ const (
 //
 // ```
 // match:
-//   path:
-//     value: "/foo"
-//   headers:
-//   - name: "version"
-//     value "v1"
+//
+//	path:
```

Comment:
Is this additional indentation required to pass the updated linter?

Reply:
I invoked golangci-lint with fix in in order to resolve the findings. I didn't add a Makefile target for it but it's quite useful. For example cluster-api has it: https://github.com/kubernetes-sigs/cluster-api/blob/1917d52af16a7df70c5e93c9bb0820d5c04f972f/Makefile#L478

Reply:
@robscott I can add the lint-fix target. No problem at all. 

Given how gofmt insists on the current format is incorrect and want's the code reformatted, maybe we should just disable that specific linter and undo the changes? Whats your opinion on that?

Reply:
I'm not sure, disabling all of gofmt for this relatively niche use case does seem unfortunate. We could possibly make an exception like this for gofmt that only applied to our API type definitions: https://github.com/kubernetes-sigs/gateway-api/blob/main/.golangci.yml#L42. Interested in what others think here as well.

/cc @shaneutt @youngnick

Reply:
To ensure the changes to api where from the golangci-linter I checked out main and ran it again.

```shell
$ git checkout main apis
Updated 6 paths from 6e0a1739
$ make lint
hack/verify-golint.sh
(warnings omitted...)
apis/v1beta1/httproute_types.go:343: File is not `gofmt`-ed with `-s` (gofmt)
// * ":method" - ":" is an invalid character. This means that HTTP/2 pseudo
//   headers are not currently supported by this type.
// * "/invalid" - "/" is an invalid character
apis/v1beta1/shared_types.go:337: File is not `gofmt`-ed with `-s` (gofmt)
// 1. IPs are not allowed.
// 2. A hostname may be prefixed with a wildcard label (`*.`). The wildcard
//    label must appear by itself as the first label.
apis/v1alpha2/grpcroute_types.go:277: File is not `gofmt`-ed with `-s` (gofmt)
// - method:
//    type: Exact
//    service: "foo"
//   headers:
make: *** [Makefile:118: lint] Error 1
$ make lint-fix
(warnings omitted...)
$ make lint
hack/verify-golint.sh
(warnings omitted...)
```

Reply:
So the gofmt linter fails as is in main.

Reply:
I wonder if there's some prior art somewhere nearby in the sigs for avoiding this yaml problem :thinking:

Reply:
There doesn't seem to be any specific configuration to manage this. For example: 
https://github.com/kubernetes-sigs/kubebuilder/blob/master/.golangci.yml
https://github.com/kubernetes-sigs/cluster-api/blob/main/.golangci.yml

I assume they follow the linters suggestions

Reply:
https://github.com/kubernetes-sigs/kubebuilder/blob/f4dbb5c82e436c6943d81abb2bf0b5b2b3478921/docs/book/src/cronjob-tutorial/running-webhook.md?plain=1#L37-L39

This is maybe going a different way. But I like this approach by Kubebuilder.

Reply:
I think we're stuck between two less than ideal options:

**1. Don't run gofmt and goimports on our types files**
This can be accomplished pretty easily with the following addition to `exclude-rules`:
```
    - path: _types\.go
      linters:
        - gofmt
        - goimports
```

It allows us to keep inline YAML examples in our godocs and therefore generated reference docs. 

**2. Remove YAML examples from inline docs**
Although this may make our generated reference docs slightly harder to follow, it would ensure that our types files were well formatted.

**3. Don't upgrade golangci-lint yet**
This appears to pass with our current generation of golangci-lint which is only slightly older than this one.


At this point, I don't love any of the options, but am slightly leaning towards 3 here unless we can find a better alternative than 1 or 2.

Reply:
I think one thing to remember here is that many editors will format on save for Go. I can't remember explicitly enabling it (I might be wrong here) in IDEA but I nevertheless I have it. If the linter expects a certain indentation, but your tooling automatically reformats the code upon save, it's going to be a pain for contributors. I think there is a risk of many prs failing the lint step in the future. 

Ultimately it's your call. I can fix 1, 2 or close it for 3.

Reply:
it's possible to embed yaml, just the entire block needs to be indented to conform to gofmt's notion of a code block

Reply:
From https://go.dev/doc/comment
indenting all the yaml by a tab should make it a preformatted code block

Reply:
Awesome find @seankhliao! Thank you so for your help here. I've updated the indents with tabs and at least locally it doesn't reformat or linting fail.